subsys/bluetooth/host/att.c

```diff
@@ -214,17 +214,27 @@ static void bt_att_disconnected(struct bt_l2cap_chan *chan);
 struct net_buf *bt_att_create_rsp_pdu(struct bt_att_chan *chan,
 				      uint8_t op, size_t len);
 
+static void bt_att_sent(struct bt_l2cap_chan *ch);
+
 void att_sent(struct bt_conn *conn, void *user_data)
 {
 	struct bt_att_tx_meta_data *data = user_data;
 	struct bt_att_chan *att_chan = data->att_chan;
 	struct bt_l2cap_chan *chan = &att_chan->chan.chan;
 
+	if (IS_ENABLED(CONFIG_BT_EATT)) {
+		__ASSERT_NO_MSG(!bt_att_is_enhanced(att_chan));
+	}
```

Comment:
`bt_att_is_enhanced` has the if-def-optimization internally. 
```suggestion
	__ASSERT_NO_MSG(!bt_att_is_enhanced(att_chan));
```